### PR TITLE
Collimat flag remove

### PR DESCRIPTION
**Pull request by kyrsjo**
Removing the COLLIMAT compilation flag.

NOTE: In progress, do not merge yet.

### CODE REVIEW

File: source/track_thin.f90 [hunk 1]
```diff
@@ -1159,7 +1152,7 @@ subroutine thin4d(nthinerr)
 
 630 continue
 
-#if defined(ROOT) && !defined(COLLIMAT)
```
Comment (kyrsjo) [inline source/track_thin.f90]: WTF? @JamesMolson

File: source/track_thin.f90 [hunk 2]
```diff
@@ -2154,11 +2143,11 @@ subroutine thin6d(nthinerr)
...
 650 continue !END loop over structure elements
 
-#ifdef COLLIMAT
-    call collimate_end_turn
-#endif
+    if (do_coll) then
+      call collimate_end_turn
+    endif
 
-#if defined(ROOT) && !defined(COLLIMAT)
```
Comment (kyrsjo) [inline source/track_thin.f90]: Huh? @JamesMolson

**Comment by kyrsjo** [comment]
Done! Seems to work perfectly. Veronica also added 1 very relevant new test (PR #603).

**Comment by kyrsjo** [comment]
Thanks! It would be great if @JamesMolson could take a look at the ROOT / COLLIMAT build flag stuff that I tagged; the logic of the flag combined with the logic inside the IFDEF looked strange.

**Comment by JamesMolson** [comment]
@kyrsjo The logic is totally fine

File: test/CMakeLists.txt [hunk 3]
```diff
@@ -174,14 +173,14 @@ endif()
 
 #Collimation tests (needs backtrk)
 #use extra checks on fort.999 and other collimation loss files
-if(COLLIMAT AND BACKTRK)
+if(BACKTRK)
```
Comment (JamesMolson) [inline test/CMakeLists.txt]: this flag does not exist

File: source/aperture.f90 [hunk 4]
```diff
@@ -1021,20 +1006,20 @@ subroutine lostpart(turn, i, ix, llost, nthinerr)
           lparID = .false.
           jjx=1
 
-#ifdef COLLIMAT
-          do jj=1,npart
```
Comment (JamesMolson) [inline source/aperture.f90]: @amereghe This bit

Comment (amereghe) [reply source/aperture.f90]: https://github.com/SixTrack/SixTrack/blame/master/source/aperture.f90

**Comment by amereghe** [comment]
to avoid mistakes, let's block run if post-pr is there together with FLUKA / COLLIMAT / SCATTER blocks, to avoid inconsistencies in `fort.10`

File: source/mainarrays.f90 [hunk 5]
```diff
@@ -56,9 +54,7 @@ subroutine allocate_arrays
 #ifdef CR
   call cr_expand_arrays(npart)
 #endif
-#ifdef COLLIMAT
   call collimation_allocate_arrays
-#endif
```
Comment (vkbo) [inline source/mainarrays.f90]: Hmm, this is not a good idea. Collimation allocates a lot of arrays. Should probably find a way around this.

**Comment by kyrsjo** [comment]
Done! Please merge.

File: source/mainarrays.f90 (hunk 5 above)
Comment (vkbo) [reply source/mainarrays.f90]: @kyrsjo There's still this ...

**Comment by vkbo** [comment]
Not yet! See my comment above.

File: source/mainarrays.f90 (hunk 5 above)
Comment (kyrsjo) [reply source/mainarrays.f90]: Could we call that from inside the COLL block input parsing maybe?

Comment (vkbo) [reply source/mainarrays.f90]: yeah, we could probably call the alloc routine with sizes 1 (if unallocated causes stuff to explode) and then call expand after a coll block has been seen. I.e. in daten when the openBlock flag is true.

File: source/aperture.f90 (hunk 4 above)
Comment (kyrsjo) [reply source/aperture.f90]: I'll let you sort this out in a different PR :)

**Pull request merged**